src/devices/secplus_v2.c

```diff
@@ -245,15 +257,16 @@ static int secplus_v2_callback(r_device *decoder, bitbuffer_t *bitbuffer)
 {
     unsigned search_index = 0;
     bitbuffer_t bits = {0};
-    // int i            = 0;
 
-    //bitbuffer_t bits_1    = {0};
-    bitbuffer_t fixed_1   = {0};
-    uint8_t rolling_1[16] = {0};
+    struct timeval tv_r1;
+    struct timeval tv_r2;
+    struct timeval tv_now;
 
-    //bitbuffer_t bits_2    = {0};
-    bitbuffer_t fixed_2   = {0};
-    uint8_t rolling_2[16] = {0};
+    struct secplus_v2_stateful_context *context = decoder->decode_ctx;
```

Comment:
This is pretty complicated (which is how it is), but I am not finding comments that explain the grand plan, what info is stored, how the transmitter works, etc.  I am not finding an explanation of whether the two packets are tied together or whether a 1st packet from one can be used with a 2nd from another.   I'm not saying your code is wrong.  Rather, I'm asking for the reader to be led to understand all the hard parts and be able to follow an argument that it's correct.